app/services/file_workflow_service.py

```
@@ -1155,6 +1164,269 @@ def _recursive_scandir(self, path: Path) -> Iterator[os.DirEntry]:
+                    if old_storage_ref != storage_reference:
+                        logger.info(
+                            "Self-healing: file ID %s moved/re-uploaded (new ref=%s, old ref=%s)",
+                            db_id_match,
+                            storage_reference,
+                            old_storage_ref,
+                        )
+                        matched_entry.file_path = storage_reference
+                        
+                        # Update the reference in our local cache dictionary
+                        if old_storage_ref in existing_by_ref:
+                            del existing_by_ref[old_storage_ref]
```

Ruff (0.15.15)

[warning] Use `pop` instead of `key in dict` followed by `del dict[key]`

Replace `if` statement with `.pop(..., None)`

(RUF051)